module.go

```diff
@@ -122,14 +122,25 @@ func (m *Module) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	wafresponse := out.WAFResponse
 	switch {
 	case m.config.IsAllowCode(int(wafresponse)):
-		// continue with normal request
+		// Continue with normal request
 		m.handler.ServeHTTP(rw, req)
 	case m.config.IsBlockCode(int(wafresponse)):
 		status := int(wafresponse)
+
+		// Only redirect if it is a redirect status (3xx) AND there is a redirect URL
+		if status >= 300 && status <= 399 {
+			redirect := req.Header.Get("X-Sigsci-Redirect")
```

Comment:
May be use `out.RequestHeaders` to read `X-Sigsci-Redirect` value ?

Reply:
Yes. I think this is a better place to read this, but it is in `[][2]string` format vs the `map[string][]string` format. Since we already are looping through that to add them to `req.Header`, I did not want to have to loop through it again and cause a potential perf issue. Instead I made sure that `X-Sigsci-Redirect` is deleted from the `req.Header` so that no client could supply this and then only add it if it comes from the agent in `out.RequestHeaders`. This seems a bit more efficient.